zilliqa/src/consensus.rs

```diff
@@ -1081,7 +1076,7 @@ impl Consensus {
                 return None;
             }
             let head_block = self.head_block();
-            let leader = self.leader_at_block(&head_block, view).unwrap();
+            let leader = self.leader_at_block(&head_block).unwrap();
```

Comment:
Shouldn't we rather use the `leaderAtView()` function of the `Deposit` contract and pass in the `view` argument? The `view` parameter of the `leader_for_view()` function could refer to an arbitrary view number, not necessarily the current view.